core/server/sources_test.go

```diff
@@ -144,3 +144,152 @@ func TestListHelmRepositories(t *testing.T) {
 }
+
+func TestCreateHelmChart(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	ctx := context.Background()
+
+	c, cleanup := makeGRPCServer(k8sEnv.Rest, t)
+	defer cleanup()
+
+	_, k, err := kube.NewKubeHTTPClientWithConfig(k8sEnv.Rest, "")
+	g.Expect(err).NotTo(HaveOccurred())
+
+	t.Run("with app association", func(t *testing.T) {
+		ns := newNamespace(ctx, k, g)
+
+		r := &pb.AddHelmChartReq{
+			Name:      "mykustomization",
+			Namespace: ns.Name,
+			AppName:   "someapp",
+			SourceRef: &pb.SourceRef{
+				Kind: pb.SourceRef_HelmRepository,
+				Name: "myhelmrepository",
+			},
+			Chart:    "mychart",
+			Version:  "v0.0.0",
+			Interval: &pb.Interval{Minutes: 1},
+		}
+
+		res, err := c.AddHelmChart(ctx, r)
+		g.Expect(err).NotTo(HaveOccurred())
+
+		g.Expect(res.Success).To(BeTrue())
+
+		actual := &sourcev1beta1.HelmChart{}
+
+		g.Expect(k.Get(ctx, types.NamespacedName{Name: r.Name, Namespace: ns.Name}, actual)).To(Succeed())
+
+		expected := stypes.ProtoToHelmChart(r)
+
+		opt := cmpopts.IgnoreFields(sourcev1beta1.HelmChart{}, diffIgnoredFields...)
+		diff := cmp.Diff(*actual, expected, opt)
+
+		if diff != "" {
+			t.Error(fmt.Errorf("(-actual +expected):\n%s", diff))
+		}
+	})
+
+	t.Run("no app association", func(t *testing.T) {
+		ns := newNamespace(ctx, k, g)
+
+		r := &pb.AddHelmChartReq{
+			Name:      "mykustomization",
+			Namespace: ns.Name,
+			AppName:   "",
+			SourceRef: &pb.SourceRef{
+				Kind: pb.SourceRef_HelmRepository,
+				Name: "myhelmrepository",
+			},
+			Chart:    "mychart",
+			Version:  "v0.0.0",
+			Interval: nil,
+		}
+
+		res, err := c.AddHelmChart(ctx, r)
+		g.Expect(err).NotTo(HaveOccurred())
+
+		g.Expect(res.Success).To(BeTrue())
+
+		actual := &sourcev1beta1.HelmChart{}
+
+		g.Expect(k.Get(ctx, types.NamespacedName{Name: r.Name, Namespace: ns.Name}, actual)).To(Succeed())
+
+		expected := stypes.ProtoToHelmChart(r)
+
+		opt := cmpopts.IgnoreFields(sourcev1beta1.HelmChart{}, diffIgnoredFields...)
+		diff := cmp.Diff(*actual, expected, opt)
+
+		if diff != "" {
+			t.Error(fmt.Errorf("(-actual +expected):\n%s", diff))
+		}
+
+		g.Expect(actual.Labels["app.kubernetes.io/part-of"]).To(Equal(""))
+	})
+}
+
+func TestListHelmCharts(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	ctx := context.Background()
+
+	c, cleanup := makeGRPCServer(k8sEnv.Rest, t)
+	defer cleanup()
+
+	_, k, err := kube.NewKubeHTTPClientWithConfig(k8sEnv.Rest, "")
+	g.Expect(err).NotTo(HaveOccurred())
+
+	appName := "myapp"
+	ns := newNamespace(ctx, k, g)
+
+	r := &pb.AddHelmChartReq{
+		Name:      "myhelmrepository",
+		Namespace: ns.Name,
+		AppName:   appName,
+		SourceRef: &pb.SourceRef{
+			Kind: pb.SourceRef_HelmRepository,
+			Name: "myhelmrepository",
+		},
+		Chart:    "mychart",
+		Version:  "v0.0.0",
+		Interval: &pb.Interval{Minutes: 1},
+	}
+
+	addRes, err := c.AddHelmChart(ctx, r)
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(addRes.Success).To(BeTrue())
+
+	unAssociatedKustomizationReq := &pb.AddHelmChartReq{
```

Comment:
Variable name :). Also, we are repeating ourselves here in a couple places. Can anything be refactored in a helper func?